src/EditorFeatures/Core/Diagnostics/RawDiagnosticTaggerConfiguration.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.Diagnostics;
+
+internal enum RawDiagnosticTaggerConfiguration
+{
+    Compiler = 1 << 0,
+    Analyzer = 1 << 1,
+    Syntax = 1 << 2,
+    Semantic = 1 << 3,
```

Comment:
I believe this enum can be removed and we can instead use `DiagnosticKinds` enum added in https://github.com/dotnet/roslyn/pull/65583/files#diff-8c48368ac3f651944c8095452e74ce75fb1f94c3f4649e5aa8632ea28b471cc5

I prefer the enum semantics in `DiagnosticKinds` over the ones here in `RawDiagnosticTaggerConfiguration` here. For example, just specifying `RawDiagnosticTaggerConfiguration.Compiler` seems to include a larger set (compiler syntax + semantic) then specifying `RawDiagnosticTaggerConfiguration.Compiler | RawDiagnosticTaggerConfiguration.Syntax` (compiler syntax). Intuitively, an `|` operation should move towards a bigger set.

Reply:
I'm fine with that. Though note my semantics were that | makes a bigger set. E.g. Compiler on its own would give nothing, since you didn't ask for syntax or semantics. 

But I don't care :-). I just wrote this so if have something to compile against while I wrote this draft :-)

Reply:
Moved to your enum.